bot/src/main/java/com/community/tools/service/TaskStatusServiceRestImpl.java

```diff
@@ -0,0 +1,172 @@
+        currentStatus = TaskStatus.NEW.getDescription();
+      } else {
+        currentStatus = repo.getLabels().get(0);
+      }
+      return new TaskNameAndStatus(repo.getTaskName(), repo.getPullUrl(),
+          currentStatus);
+    }).collect(Collectors.toList());
+  }
+
+  /**
+   * Checks if the user is active.
+   *
+   * @param user The user for whom the check is performed.
+   * @return true if the user is active, otherwise false.
+   */
+  private boolean isActiveUser(GithubUserDto user) {
+    LocalDate lastCommitDate = user.getLastCommit();
+    boolean allTasksApproved = areAllUserTasksApproved(user);
+
+    if (lastCommitDate == null || daysBetween(lastCommitDate, LocalDate.now()) >= 7) {
+      return false;
+    }
+
+    return allTasksApproved;
+  }
+
+  /**
+   * Checks if all user tasks are approved and have reviews.
+   *
+   * @param user The user for whom the check is performed.
+   * @return true if all tasks are approved and have reviews, otherwise false.
+   */
+  private boolean areAllUserTasksApproved(GithubUserDto user) {
+    List<TaskNameAndStatus> taskStatuses = getAllTaskNameAndStatusesForEachUser(user);
+
+    for (TaskNameAndStatus taskStatus : taskStatuses) {
+      String status = taskStatus.getTaskStatus();
+      if (status.equals(TaskStatus.UNDEFINED.getDescription())) {
+        return false;
+      }
+
+      if (!hasReviews(taskStatus, user)) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  /**
+   * Checks if a task has reviews.
+   *
+   * @param taskStatus The task and its status.
+   * @param user       The user for whom the check is performed.
+   * @return true if the task has reviews, otherwise false.
+   */
+  private boolean hasReviews(TaskNameAndStatus taskStatus, GithubUserDto user) {
+    for (GithubRepositoryDto repository : user.getRepositories()) {
+      if (isMatchingTask(repository, taskStatus)) {
+        if (hasReviewLabel(repository)) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Checks if a task matches a repository.
+   *
+   * @param repository The repository to check.
+   * @param taskStatus The task and its status.
+   * @return true if the task matches the repository, otherwise false.
+   */
+  private boolean isMatchingTask(GithubRepositoryDto repository, TaskNameAndStatus taskStatus) {
+    return repository.getTaskName().equals(taskStatus.getTaskName());
+  }
+
+  /**
+   * Checks if a GitHub repository has a label containing "failure".
+   *
+   * @param repository The GitHub repository to check for labels.
+   * @return true if the repository has a label not containing "failure", otherwise false.
+   */
+  private boolean hasReviewLabel(GithubRepositoryDto repository) {
+    List<String> labels = repository.getLabels();
+    if (labels != null && !labels.isEmpty()) {
+      for (String label : labels) {
+        if (!label.contains("failure")) {
+          return true;
+        }
```

Comment:
Can we just check for null here and then do List.contains("failure")? Or in case if failure must be a part of string we can do something like
`stringList.stream().anyMatch(s -> s.contains("abc"));`

Reply:
This class was entirely pre-existing logic from main branch lifted out from TaskStatusService.java file, which was turned into an interface during the refactoring process, those explicit REST api calls were only used once on application startup to populate task status table with semi-relevant information, the whole Label part of determining task status was unclear to me, and from how our first part of the traineeship was conducted, it seems to no longer be in use. Since we've clarified in todays meeting that I am free to suggest how we change the task status based on github events, this logic is no longer relevant at all and thus I decided to remove it entirely in this latest revision. I guess this also partly adresses the javadoc use, I hope that in the rest of the pull request they are relevant enough to keep)